src/app/components/Page.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+'use client';
+
+import { useAuthContext } from '@/app/hooks/AuthProvider';
+import { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import { DnaLoader, css } from '@icgc-argo/uikit';
+import { redirect } from 'next/navigation';
+import useUserRole, { UserRoleList } from '../hooks/useUserRole';
+
+export type PageWithPermissionProps = { permissions: { shortName?: string } };
+
+type AcceptedRole = keyof UserRoleList;
+
+// Component type for this project because we use the Emotion compiler
+type JSXComponent<T extends object> = (args: T) => EmotionJSX.Element;
+
+// returns Comp or unauthorized based on acceptedRoles
+// params value and types are based on dynamic route of page
+export const pageWithPermissions =
+	// eslint-disable-next-line react/display-name
```

Comment:
This linter comment might not be needed now, that's why I was refactoring the returned anonymous function as a named function.